app/validation/validator.py

```diff
@@ -659,11 +659,15 @@ def validate_mutually_exclusive(self, question):
         errors = []
 
         if question['type'] == 'MutuallyExclusive':
-            for answer in question['answers']:
-                if answer['mandatory'] is True:
-                    errors.append(self._error_message('MutuallyExclusive question type cannot contain mandatory answers.'))
+            answers = question['answers']
 
-            if question['answers'][-1]['type'] != 'Checkbox':
+            if any(answer['mandatory'] is True for answer in answers):
+                errors.append(self._error_message('MutuallyExclusive question type cannot contain mandatory answers.'))
+
+            if len(answers) == 3 and not any('parent_answer_id' in answer for answer in answers):
+                errors.append(self._error_message('Too many answers have been provided.'))
```

Comment:
A little confusing that half the validation is done by limiting `"minItems": 2` and `"maxItems": 3` in `schemas/questions/mutually_exclusive.json` so it is probably worth adding a little comment to explain that.

Reply:
Yeh i don't understand why this is like this from reading the code